src/software-management/self-update.md

```diff
@@ -0,0 +1,101 @@
+
+# Introduction
+
+
+Thin-edge self-update allows a Cloud Operator to update one thin-edge version that is installed on a device to a newer version, via the Cloud connection. The feature is referred as self-update cause that update process is managed by thin-edge own's Software Management.
+
+# Scope
+  * The self-update makes use of usual thin-edge Software Management plugins (e.g. "APT"). So is not absolutely fail-safe, instead of that it is the foundation for it.<br/>
+    For absolutely fail-safe self-update the device would need to provide and use some transaction-safe package manager and some powerfail-safe filesystem.
+     
+  * Consequences: 
+     * Self-update can fail and could left the system in an unstable state. The device might be disconnected from Cloud.
+     * Potential fail reasons (without claim of completeness):
+       * No rollback of already processes package of one self-update request, when some subsequent package of same request fails.
+         <br/>-> leads to partially executed update. 
+       * Power-fail during update.
+         <br/>-> leads to partially executed update
+         <br/>-> can corrupt the package currently about to be updated
+         <br/>-> can corrupt the filesystem
+       * Installation of incompatibel set of thin-edge package versions (e.g. SM Agent of one release, and Mapper of another release).
+         <br/>-> thin-edge function might not given, when signifcant change happened between both release
+       * Tool(s) required by thin-edge installation procedure missing on the device (e.g. 'grep').
+         <br/>-> according package installation will stop.
+         
+  * Measures for more robusness:
+    * There are plans for thin-edge packages to reduce risks for installation failures:
+      <br/>TODO: For both, add some reference (some ‚ticket, discussion, ...)
+      1. Providing one thin-edge core package that contains everything needed for self-update (assumedly "Mapper", "SM Agent" and "SM Plugin"). 
```

Comment:
Leaving the `tedge-self-updater` separated out of this core package might help in avoiding certain failure scenarios. I've documented those concerns in this [comment](https://github.com/thin-edge/thin-edge.io-specs/pull/20#issuecomment-1049676234).